trackintel/analysis/location_identification.py

```diff
@@ -0,0 +1,214 @@
+        groupby_loc = ["location_id"]
+    else:
+        raise ValueError(f"Unknown agg_level '{agg_level}' use instead {{'user', 'dataset'}}.")
+    loc = sps.groupby(groupby_loc).agg({"started_at": [min, "count"], "finished_at": max, "duration": sum})
+    loc.columns = loc.columns.droplevel(0)  # remove possible multi-index
+    loc.rename(columns={"min": "started_at", "max": "finished_at", "sum": "duration"}, inplace=True)
+    loc["period"] = loc["finished_at"] - loc["started_at"]
+    loc_sp = loc["count"] >= thresh_sp_at_loc
+    # how should we handle the timedeltas as input to our function
+    # are both integers or do we accept other values?
+    if isinstance(thresh_loc_time, int):
+        thresh_loc_time = pd.to_timedelta(thresh_loc_time, unit="h")
+    loc_time = loc["duration"] >= thresh_loc_time
+    loc_period = loc["period"] >= thresh_loc_period
+    loc_filter_agg = loc_sp & loc_time & loc_period
+    loc_filter_agg.rename("loc_filter", inplace=True)  # rename for merging
+    loc_filter = pd.merge(sps[groupby_loc], loc_filter_agg.reset_index(), on=groupby_loc, how="left")["loc_filter"]
+
+    total_filter = user_filter & loc_filter
+
+    return total_filter
+
+
+def freq_recipe(sps, *labels):
+    """Generate a location activity per user by assigning the most visited location the label "home"
+    and the second most visited location the label "work". The remaining locations get no label.
+
+    Parameters
+    ----------
+    sps : GeoDataFrame (as trackintel staypoints)
+        Staypoints with the column "location_id".
+    labels : collection of str, default ("home", "work")
+        Labels in decreasing time of activity.
+
+    Returns
+    -------
+    GeoDataFrame
+        with column "activity_label".
+
+    Examples
+    --------
+    >> staypoints["activity"] = ti.analysis.freq_recipe(staypoints, "home", "work")
+    """
+    sps = sps.copy()
+    if not labels:
+        labels = ("home", "work")
+    for name, group in sps.groupby("user_id"):
+        if "duration" not in group.columns:
+            group["duration"] = group["finished_at"] - group["started_at"]
+        # pandas keeps inner order or groups
+        sps.loc[sps["user_id"] == name, "activity_label"] = _freq_transform(group, *labels)
+    return sps
+
+
+def _freq_transform(group, *labels):
+    """Transform function that assigns the longest visited data point the labels in order.
+
+    Parameters
+    ----------
+    group : DataFrame
+
+    Returns
+    -------
+    pd.Series
+        dtype : `object`
+    """
+    group_agg = group.groupby("location_id").agg({"duration": sum})
+    group_agg["activity_label"] = _freq_assign(group_agg["duration"], *labels)
+    group_merge = pd.merge(
+        group["location_id"], group_agg["activity_label"], how="left", left_on="location_id", right_index=True
+    )
+    return group_merge["activity_label"]
+
+
+def _freq_assign(duration, *labels):
+    """Assign labels to array with longest duration.
```

Comment:
Is this `longest duration` or `k longest duration` (here 1st is home and 2nd is work)?